paper.md

```diff
@@ -0,0 +1,55 @@
+
+# Statement of Need
+
+Spatial filtering is commonly used as a scientific tool for analyzing gridded data. An example of an existing spatial filtering tool in python is `SciPy`'s [@2020SciPy-NMeth] `ndimage.gaussian_filter` function, implemented as a sequence of convolution filters. While being a valuable tool for image processing (or blurring), `SciPy`'s Gaussian filter is of limited use for GCM data; it assumes a regular and rectangular Cartesian grid, employs a simple boundary condition, and the definitions of filter scale and shape have little or no flexibility. The python package `GCM-Filters` is specificially designed to filter GCM data, and seeks to solve a number of challenges for the user:
+
+1. GCM data comes on irregular curvilinear grids with spatially varying grid-cell geometry.
+2. Continental boundaries require careful and special treatment when filtering ocean GCM output.
+3. Earth Science applications benefit from configurable filters, where the definition of filter scale and shape is flexible.
+4. GCM output often comes in very large out-of-memory datasets.
+
+The `GCM-Filters` algorithm [@grooms2021diffusion] applies a discrete Laplacian to smooth a field through an iterative process that resembles diffusion. The discrete Laplacian takes into account the varying grid-cell geometry and uses a no-flux boundary condition, mimicking how diffusion is internally implemented in GCMs. The no-flux boundary conditions ensures that the filter preserves the integral: $\int_{\Omega} \bar{f}(x,y) \,dx dy = \int_{\Omega} f (x,y)\, dx dy$, where $f$ is the original field, $\bar{f}$ the filtered field, and $\Omega$ the ocean domain. Conservation of the integral is a desirable filter property for many physical quantities, for example energy or ocean salinity. More details on the filter properties can be found in @grooms2021diffusion.
+
+The main `GCM-Filters` class that the user will interface with is the `gcm_filters.Filter` object. When creating a filter object, the user specifies how they want to smooth their data, including the desired filter shape and filter scale. At this stage, the user also picks the grid type that matches their GCM data, given a predefined list of grid types. Each grid type has an associated discrete Laplacian, and requires different *grid variables* that the user must provide (the latter are usually available to the user as part of the GCM output). Currently, `GCM-Filters` provides a number of different grid types and associated discrete Laplacians:
+
+* Grid types with **scalar Laplacians** that can be used for filtering scalar fields, for example temperature or vorticity (see \autoref{fig1}). The currently implemented grid types are compatible with different ocean GCM grids including MOM5, MOM6 and the POP tripole grid.
```

Comment:
Do we want to cite MOM / POP references here?

Reply:
Yes, referencing the models (MOM5, MOM6, POP2, MITgcm) would be nice. Does anybody see a straightforward way to reference these? As of now, we have:
* MITgcm: [Campin et al., 2021](https://zenodo.org/record/4968496#.YTFQbhNKiL5) (Zenodo doi); alternatively: [Adcroft et al., 2018](https://dspace.mit.edu/handle/1721.1/117188) (MITgcm user manual, Zenodo doi) or Marshall et al., [1997a](https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/96JC02776), [1997b](https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/96JC02775) (both JGR Oceans)
* MOM6: [Adcroft et al., 2019](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2019MS001726) (JAMES paper)
* MOM5: [github page](https://github.com/mom-ocean/MOM5), no doi
* POP2: [github page](https://github.com/ESCOMP/POP2-CESM), no doi